### PR TITLE
[C5][environment] Adding @OPTIONS annotation via codegen 

### PR DESCRIPTION
**Pull request by RenukaFernando**
### Proposed changes in this pull request

- Add 'autoEnableOptions' in swagger yaml file in publisher, store, webserver
- Delete produces application/json and text/html in webserver swagger yaml file.
- Codegen using swagger2msf4j in publisher, store, webserver
- Manually edit webserver PublisherAPI generated file
  - Change `@Path(".[\\d]+/publisher")` to `@Path(".*")`

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/ApiResponseMessage.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    public static final int WARNING = 2;
+    public static final int INFO = 3;
+    public static final int OK = 4;
+    public static final int TOO_BUSY = 5;
+
+    int code;
+    String type;
+    String message;
+
+    public ApiResponseMessage(){}
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/ApiResponseMessage.java]: Formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/ApiResponseMessage.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    public ApiResponseMessage(int code, String message){
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/ApiResponseMessage.java]: Formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/ApiOriginFilter.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package org.wso2.carbon.apimgt.rest.api.webserver;
+
+import java.io.IOException;
+
+import javax.servlet.*;
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/ApiOriginFilter.java]: Add all the importing packages/classes without using * for imports.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/PublisherApi.java [hunk 4]
```diff
@@ -34,30 +36,32 @@
     service = Microservice.class,
     immediate = true
 )
+//@Path(".[\\d]+/publisher")
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/PublisherApi.java]: Remove the commented code block

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/pom.xml [hunk 5]
```diff
@@ -127,14 +127,15 @@
...
+            <plugin>
                 <groupId>org.wso2.maven.plugins</groupId>
                 <artifactId>swagger2msf4j-maven-plugin</artifactId>
                 <version>1.0-SNAPSHOT</version>
                 <configuration>
                     <inputSpec>${project.basedir}/src/main/resources/publisher-api.yaml</inputSpec>
+                    <autoEnableOptions>true</autoEnableOptions>
                 </configuration>
-            </plugin> -->
+            </plugin>
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/pom.xml]: Please keep commenting the swagger2msf4j-maven-plugin.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/main/resources/webserver-api.yaml [hunk 6]
```diff
@@ -90,8 +90,8 @@ paths:
         - $ref: '#/parameters/Accept'
         - $ref: '#/parameters/If-None-Match'
         - $ref: '#/parameters/If-Modified-Since'
-      produces:
-        - text/html; charset=utf-8
+#      produces:
+#        - text/html; charset=utf-8
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/main/resources/webserver-api.yaml]: Any reason to keep the commented code ?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/ApiResponseMessage.java (hunk 1 above)
Comment (RenukaFernando) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/ApiResponseMessage.java]: Fixed with https://github.com/RenukaFernando/carbon-apimgt/commit/88634c560fb8dfbf87211a105d87445605d8592f

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/ApiResponseMessage.java (hunk 2 above)
Comment (RenukaFernando) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/ApiResponseMessage.java]: Fixed with https://github.com/RenukaFernando/carbon-apimgt/commit/88634c560fb8dfbf87211a105d87445605d8592f

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/ApiOriginFilter.java (hunk 3 above)
Comment (RenukaFernando) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/ApiOriginFilter.java]: Fixed with https://github.com/RenukaFernando/carbon-apimgt/commit/88634c560fb8dfbf87211a105d87445605d8592f

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/PublisherApi.java (hunk 4 above)
Comment (RenukaFernando) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/PublisherApi.java]: Fixed with https://github.com/RenukaFernando/carbon-apimgt/commit/88634c560fb8dfbf87211a105d87445605d8592f

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/pom.xml (hunk 5 above)
Comment (RenukaFernando) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/pom.xml]: Fixed with https://github.com/RenukaFernando/carbon-apimgt/commit/aa83fd7a66865b2dd15f6041bed648361f036494

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/main/resources/webserver-api.yaml (hunk 6 above)
Comment (RenukaFernando) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/main/resources/webserver-api.yaml]: Fixed with https://github.com/RenukaFernando/carbon-apimgt/commit/fe0833d6cecaf9b8d24701dc2be2157ffdf245b9

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/ApiResponseMessage.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    int code;
+    String type;
+    String message;
+
+    public ApiResponseMessage() {
+    }
+
+    public ApiResponseMessage(int code, String message) {
+        this.code = code;
+        switch(code){
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/gen/java/org/wso2/carbon/apimgt/rest/api/webserver/ApiResponseMessage.java]: Formatting issue

**Pull request merged**